TheThingsNetwork/lorawan-stack
# console: Use the events WebSocket API

#### Summary

This PR replaces server side events with single socket connection using WebSocket API

#### Changes

Refactored [streaming module of the JS SDK](https://github.com/TheThingsNetwork/lorawan-stack/tree/v3.28/sdk/js/src/api/stream). Now we are using only `/api/v3/console/internal/events/` endpoint for fetching events for all entities, as opposed to using multiple per entity type. Using the native [WebSocket API](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket).

#### Testing

1. Login in the console.
2. Go to a details page consisting live data, ex. `/console/applications/:appId`, `/console/gateways/:gatewayid` etc.
3. Check if the fetched data is correct and in real time.
4. (Extra) Check in Developers console -> Network tab if we send/receive data correctly via the web socket.

#### Notes for Reviewers

Initialy like a temporary fix to test this(until it is being implemented, I already contacted @adriansmares), you need to hardcode this line into `pkg/webui/csp.go` file and before line 108 so you could establish a connection with the web socket:
`	
csp.ConnectionSource = append(csp.ConnectionSource, "wss://tti.staging1.cloud.thethings.industries/api/v3/console/internal/events/", "ws://localhost:1885/api/v3/console/internal/events/", 
"ws://localhost:8080/api/v3/console/internal/events/")
`

#### Checklist
